lib/reconcile/astarte_generic_backend.go

```diff
@@ -78,6 +79,12 @@ func EnsureAstarteGenericBackendWithCustomProbe(cr *apiv1alpha1.Astarte, backend
 		return err
 	}
 
+	// Good. Now, reconcile the service first of all.
+	service := &v1.Service{ObjectMeta: metav1.ObjectMeta{Name: serviceName, Namespace: cr.Namespace}}
```

Comment:
You can create this inside `createOrUpdateService` (and pass `serviceName` as parameter instead of `service`, since you already have the `cr` to take the namespace)